crates/telemetry-subscribers/src/lib.rs

```diff
@@ -353,10 +360,26 @@ impl TelemetryConfig {
 
         if config.sample_nth.is_some() {
             let sample_nth = config.sample_nth.unwrap();
-            let sampling_layer = SamplingFilter::new(sample_nth, config.target_prefix);
+            let sampling_layer = SamplingFilter::new(sample_nth, config.target_prefix.clone());
             layers.push(sampling_layer.boxed());
         }
 
+        let gas_stats_filter_prefix = config.gas_stats_filter_prefix.clone();
+        let gas_filter = filter::filter_fn(move |metadata| {
+            if let Some(ref prefix) = gas_stats_filter_prefix {
+                return metadata.target().starts_with(prefix);
+            }
+            false
+        });
```

Comment:
So looking at this you're passing in a prefix for filtering of certain log lines. I'm not quite sure i understand the motivations for doing this vs using the already existing filtering mechanisms. In other words you can do exactly the same thing today without any changes just by setting `RUST_LOG` properly

Reply:
I want to be able to add only specific lines of logging to the local files, so that we don't have storage concerns when gathering data for the analysis. These log lines will be in the sui node side of things. How would RUST_LOG allow us to do that?

Reply:
The `RUST_LOG` env var can be set with a set of [directives/filters](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html#directives) to do just what you're looking for. Eg `RUST_LOG=gas-prefix=trace,off` will only enable log lines with targets that have the prefix `gas-prefix`

Reply:
So it comes down to what your intent or goal is here. When/where do you want to be able to collect logs for gas profiling/stats? Do you want it be to run on a production node where we also want the other logs that we generally emit or would you want this only for local profiling, in which case its generally going to be ok to disable the normal logs that are emitted

Reply:
thanks @bmwill, so we could use a `gas-flag` for logging gas, and turn it on only when we are collecting gas stats. 
When we do it, we also do not log anything else. 
It's a local run, we grab the log file and it only contains the thing you care about.
That maybe a good plan (@laura-makdah?)
Out of curiosity, is there a way (easy way) to send a given "log type" (`gas-flag` here) to a specific file and everything else to the regular log?

Reply:
I would like to leave the possibility of running this while not interfering the other logs, which this prefix method here should allow for yes? I didn't know you could add a prefix to RUST_LOG like that, I am sure it will come in handy when doing local experiments and wanting to filter it further.

Reply:
> I would like to leave the possibility of running this while not interfering the other logs

if easy to do yes, otherwise it's not clear to me there is any benefit in changing the code now. The method described is good enough for us to start working on what we log right? and that is the most critical work, no?
bottom line, given the comment, I am worried we make a change that can cause issues and the benefit is not clear at this point in time

Reply:
it is already done in that way, an additional benefit of the existing way is that we filter out and don't send the gas_stats to the prometheus backend. It is important that we maintain a reasonable log volume in production and these stats would add way too much to the prometheus backend if they were not to be filtered out